PowerShellActions/PowerShellTask.cs

```diff
@@ -84,7 +84,7 @@ public bool Execute()
                 _session.Message(InstallMessage.Error, record);
             }
 
-            return !_pipeline.HadErrors && errors == null && ExitCode == 0;
+            return ((_pipeline.Error == null) || (_pipeline.Error.Count == 0)) && errors == null && ExitCode == 0;
```

Comment:
Ah.  Can you add a comment above that line to indicate the same (otherwise someone might forget and use HadErrors in the future). Otherwise all looks good.